pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -233,6 +310,11 @@ func (p *staticPolicy) RemoveContainer(s state.State, podUID string, containerNa
 		}
 	}
 
+	// If this was a last allocation from pod free reusables
+	if len(s.GetMemoryAssignments()[podUID]) == 0 {
+		delete(p.memoryToReuse, podUID)
+	}
```

Comment:
`Size` in reusables is equal to all `Size`s in initContainer blocks, so we don't have to explicitly update NodeMap after deleting `memoryToReuse` entry, it was correctly updated when initContainers were deleted